### ISSUE TITLE
supporting DataContract/DataMember/IgnoreDataMember

### ISSUE THREAD

**Issue by BrannonKing** [tomba/netserializer#66]
I attempted to write my own inheritor of IDynamicTypeSerializer for handling DataContract objects. It would be very similar to the GenericSerializer. Some things that caused me a problem:  the `Serializer.Settings` and `Serializer.GetIndirectData` are both internal, making it impossible for me to duplicate the mechanism in GenericSerializer. It would be nice if GenericSerializer wasn't sealed and had virtual methods for `GetFieldInfos` and `Handles`.

**Comment by tomba** [comment]
Yes, I never really finished the mechanism to create custom serializers. I didn't want to make all the stuff public, as I felt it's exposing too much internal data, but as you noticed, at least some of it is needed. But if I recall right, I didn't consider making GenericSerializer inheritable. I was only thinking of how a user could write  a serializer from scratch. Perhaps making GenericSerializer inheritable would be an easier step forward.

However, I haven't done any .Net development for a long time, so I don't think I'll work on this any time soon.
